shub/bootstrap.py

```diff
@@ -0,0 +1,137 @@
+from __future__ import absolute_import
+
+import os
+import shutil
+import tempfile
+import zipfile
+
+import click
+import requests
+import yaml
+from click.formatting import HelpFormatter
+from six import BytesIO
+
+from shub.exceptions import (
+    BadParameterException, NotFoundException, RemoteErrorException)
+
+
+EXAMPLE_REPO = "scrapinghub/custom-images-examples"
+AVAILABLE_PROJECTS_URL = (
+    "https://raw.githubusercontent.com/%s/master/bootstrap_projects.yml"
+    "" % EXAMPLE_REPO)
+
+HELP = """
+Through custom images, Scrapinghub allows you to run crawlers written in any
+language you want. To get you started, we prepared a few examples projects in
+different programming languages and frameworks. You can find them in our custom
+images repository at:
+
+    https://github.com/scrapinghub/custom-images-examples
+
+The 'shub bootstrap' command clones an example project to the current directory
+so that you can start hacking right away.
+
+Run
+
+    shub bootstrap -l
+
+to get a list of all available example projects, then run
+
+    shub bootstrap PROJECTNAME
+
+to clone it.
+"""
+
+SHORT_HELP = "Clone custom image example project"
+
+
+def list_projects_callback(ctx, param, value):
+    if not value or ctx.resilient_parsing:
+        return
+    projects = get_available_projects()
+    list_projects(projects)
+    ctx.exit()
+
+
+@click.command(help=HELP, short_help=SHORT_HELP)
+@click.option('-l', '--list', 'list_projects', help='list available projects',
+              is_flag=True, callback=list_projects_callback,
+              expose_value=False, is_eager=True)
+@click.argument('project')
+@click.argument('target_dir', required=False)
+def cli(project, target_dir):
+    target_dir = os.path.join(os.getcwd(), target_dir or project)
```

Comment:
1. re subdir - sounds reasonable, I don't have stronger arguments
2. re normpath - I proposed it because it provides a better user experience if target dir path is relative and contains links to parent directories. if you specify target as `../../path` it's easier for user to read `/path` rather than `/parent/current/../../path`. IMO humans should not try to evaluate final path in their heads, this is error prone. if you're concerned about symlinks let's use `os.path.realpath` instead.